components/service/glean/src/main/java/mozilla/components/service/glean/ping/PingMaker.kt

```diff
@@ -29,8 +30,22 @@ internal class PingMaker(
      * @return a string containing the date and time.
      */
     private fun getISOTimeString(): String {
-        val dateFormat = SimpleDateFormat("yyyy-MM-dd'T'HH:mmXXX", Locale.US)
-        return dateFormat.format(Date())
+        val dateFormat = SimpleDateFormat("yyyy-MM-dd'T'HH:mmZ", Locale.US)
+        val timeString = StringBuilder(dateFormat.format(Date()))
+
+        // Due to limitations of SDK version 21, there isn't a way to properly output the time
+        // offset with a ':' character:
+        // 2018-12-19T12:36-0600    --  This is what we get
+        // 2018-12-19T12:36-06:00   -- This is what GCP will expect
+        //
+        // In order to satisfy time offset requirements of GCP, we manually insert the ":" when
+        // needed, which should be for every timezone except UTC which is signified by the trailing
+        // 'Z' character
+        if (!timeString.endsWith("Z")) {
+            timeString.insert(timeString.length - 2, ":")
+        }
```

Comment:
Just FYI: In case you ever want to write code that behaves differently on various SDK levels:
* You can read the current SDK level with: https://developer.android.com/reference/android/os/Build.VERSION.html#SDK_INT
* And compare it against constants from: https://developer.android.com/reference/android/os/Build.VERSION_CODES

Here it doesn't matter I guess and we can just use the workaround on all devices.